pyramid_stocks/pyramid_stocks/routes.py

```diff
@@ -5,5 +5,6 @@ def includeme(config):
     config.add_route('portfolio', '/portfolio')
     config.add_route('auth', '/auth')
     config.add_route('stock-add', '/stock-add')
-    config.add_route('stock-detail', '/stock-detail')
+    config.add_route('detail', '/detail/{symbol}')
+    config.add_route('404', '/404')
```

Comment:
You do not need a `404` route. Pyramid automatically sends the user to a 404 page when the 404 error is raised.